rules-generated/quarkus/kafka/kafka-to-quarkus.windup.xml

```diff
@@ -0,0 +1,118 @@
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-xml,3.0.0.Final" />
+        </dependencies>
+        <sourceTechnology id="kafka" />
+        <targetTechnology id="quarkus" />
+    </metadata>
+    <rules>
+        <rule id="quarkus-kafka-client-00000">
+            <when>
+                <project>
+                    <artifact groupId="org.apache.kafka" artifactId="kafka-clients"/>
+                </project>
+            </when>
+            <perform>
+                <hint title="Replace the 'kafka-clients' dependency with Quarkus 'quarkus-kafka-client' extension" effort="1" category-id="mandatory">
+                    <message>
+                        Replace the `org.apache.kafka:kafka-clients` dependency with the Quarkus dependency `io.quarkus:quarkus-kafka-client`  
+                        Further information in the link below.
+                    </message>
+                    <link title="Quarkus - Guide" href="https://quarkus.io/guides/kafka" />
+                </hint>
+            </perform>
+        </rule>
+        <rule id="quarkus-kafka-client-00010">
+            <when>
+                <dependency groupId="org.apache.kafka" artifactId="kafka-clients"/>
+            </when>
+            <perform>
+                <hint title="Replace the 'kafka-clients' dependency with Quarkus 'quarkus-kafka-client' extension" effort="1" category-id="mandatory">
+                    <message>
+                        Replace the `org.apache.kafka:kafka-clients` dependency with the Quarkus dependency `io.quarkus:quarkus-kafka-client`  
+                        Further information in the link below.
+                    </message>
+                    <link title="Quarkus - Guide" href="https://quarkus.io/guides/kafka" />
+                </hint>
+            </perform>
+        </rule>
+        <rule id="quarkus-kafka-client-00020">
+            <when>
+                <javaclass references="org.apache.kafka.clients.admin.{fileName}" >
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Replace the 'kafka-clients' dependency with Quarkus 'quarkus-kafka-client' extension" effort="1" category-id="mandatory">
+                    <message>
+                        Replace the `org.apache.kafka:kafka-clients` dependency with the Quarkus dependency `io.quarkus:quarkus-kafka-client`  
+                        Further information in the link below.
+                    </message>
+                    <link title="Quarkus - Guide" href="https://quarkus.io/guides/kafka" />
+                </hint>
+            </perform>
+            <where param="fileName">
```

Comment:
The basic idea about having the `fileName` is to be sure there's not the name of sub-package to trigger this rule since the rule can not be sure that a sub-package of -in this case- the `org.apache.kafka.clients.admin` package is provided always by the same `org.apache.kafka:kafka-clients` artifact.
The rule just "knows" that  `org.apache.kafka.clients.admin` package is provided by the `org.apache.kafka:kafka-clients` artifact and so it tries to be as precise as possible in searching for classes from this package and no one else.